qiskit_experiments/library/characterization/resonator_spectroscopy.py

```diff
@@ -0,0 +1,176 @@
+        - sigma: The standard deviation of the pulse, the default is duration / 4.
+        - width: The width of the flat-top in the pulse, the default is 0, i.e. a Gaussian.
+
+        Args:
+            qubit: The qubit on which to run readout spectroscopy.
+            frequencies: The frequencies to scan in the experiment, in Hz.
+            backend: Optional, the backend to run the experiment on.
+            absolute: Boolean to specify if the frequencies are absolute or relative to the
+                resonator frequency in the backend.
+        """
+        super().__init__(qubit, frequencies, backend, absolute)
+        self.analysis = ResonatorSpectroscopyAnalysis()
+        self._set_analysis_data_processor()
+
+    def _set_analysis_data_processor(self):
+        """Keep the data processor consistent with the run options."""
+        processor = get_processor(
+            self.run_options.meas_level,
+            self.run_options.meas_return,
+            self.analysis.options.normalization,
+            dimensionality_reduction=ProjectorType.ABS,
+        )
+        self.analysis.set_options(data_processor=processor)
+
+    def set_run_options(self, **fields):
+        """Wrap set run options to keep the data processor consistent."""
+        super().set_run_options(**fields)
+        self._set_analysis_data_processor()
+
+    @property
+    def center_frequency(self) -> float:
+        """Returns the center frequency of the experiment.
+
+        Returns:
+            The center frequency of the experiment.
+
+        Raises:
+            QiskitError: If the experiment does not have a backend set.
+        """
+        if self.backend is None:
+            raise QiskitError("backend not set. Cannot call center_frequency.")
+
+        return self.backend.defaults().meas_freq_est[self.physical_qubits[0]]
+
+    def _meas_template_circuit(self) -> QuantumCircuit:
+        """Return the template quantum circuit."""
+        circuit = QuantumCircuit(1, 1)
+        circuit.measure(0, 0)
+
+        return circuit
+
+    def _spec_gate_schedule(
+        self, backend: Optional[Backend] = None
+    ) -> Tuple[pulse.ScheduleBlock, Parameter]:
+        """Create the spectroscopy schedule."""
+
+        qubit = self.physical_qubits[0]
+
+        freq_param = Parameter("frequency")
+
+        with pulse.build(backend=backend, name="spectroscopy") as schedule:
+            pulse.shift_frequency(freq_param, pulse.MeasureChannel(qubit))
+            pulse.play(
+                pulse.GaussianSquare(
+                    duration=self.experiment_options.duration,
+                    amp=self.experiment_options.amp,
+                    sigma=self.experiment_options.sigma,
+                    width=self.experiment_options.width,
+                ),
+                pulse.MeasureChannel(qubit),
+            )
+            pulse.acquire(self.experiment_options.duration, qubit, pulse.MemorySlot(0))
```

Comment:
Thinking about it more, I realized this running all in one job and the IBM backends can't change the demodulation frequency within a job currently, so this is analyzing at a fixed demodulation frequency while sweeping the measurement pulse frequency. For this we should use a short acquisition in order to have enough measurement bandwidth (or a custom kernel but that seems harder to do). Otherwise we will see the roll off as the drive frequency shifts away from the demodulation frequency. Ideally, the acquisition would start after the pulse rings up the resonator also.

Reply:
With this here dfe3149 we can change the duration of the acquire instruction.